test/utils/test_observable_grouping.py

```diff
@@ -77,6 +77,11 @@ def test_most_general_observable(self):
             PauliList(["XX", "YY", "ZZ"]),
             PauliList.from_symplectic(np.zeros((3, 0)), np.zeros((3, 0))),
         ),
+        (
+            (2, 1),
```

Comment:
So flipping the order of IDs, flips the order of the Paulis..I'm sure this is intended, but worth staring at for a second 👀

Reply:
Yeah, I wanted to make sure there was an actual test where things are meant to get re-ordered.